File: search_insert_position/sip.cpp
class Solution {
public:
    //I supposed to solve it by binary search like I did in Python but I am too lazy to repeat my self so I used a std function achieve the same result ;)
    int searchInsert(vector<int>& nums, int target) {
        return lower_bound(nums.begin(),nums.end(),target)-nums.begin();
    }
};

File: search_insert_position/sip.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        l = 0;
        r = len(nums)
        mid = (l+r)//2
        while(l<r):
            if target == nums[mid]:
                break;
            elif target<nums[mid]:
                r=mid
            else :
                l=mid+1
            mid = (l+r)//2
        
        return mid
